page_values.go

```diff
@@ -122,8 +122,7 @@ func (r *booleanPageValues) ReadBooleans(values []bool) (n int, err error) {
 
 func (r *booleanPageValues) ReadValues(values []Value) (n int, err error) {
 	for n < len(values) && r.offset < int(r.page.numValues) {
-		values[n] = makeValueBoolean(r.page.valueAt(r.offset))
-		values[n].columnIndex = r.page.columnIndex
+		values[n] = r.page.makeValue(r.page.valueAt(r.offset))
```

Comment:
I made a few changes to use the `makeValue` method of page types instead of repeating the assignment of column index here.